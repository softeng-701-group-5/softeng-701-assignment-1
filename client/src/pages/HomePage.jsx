import React from 'react';
import {
  makeStyles,
  Container,
  Button,
  Paper,
  Typography,
  TextField,
} from '@material-ui/core';
import { Link } from 'react-router-dom';

const useStyles = makeStyles(theme => ({
  root: {
    backgroundImage: "url('public/bg.jpg')",
  },
  containerItems: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
  paperLayout: {
    width: '50%',
    margin: '15%',
    display: 'flex',
    flexDirection: 'column',
    textAlign: 'center',
    alignItems: 'center',
  },
  buttonLogin: {
    margin: '5%',
  },
  textInput: {
    width: '75%',
  },
  title: {
    margin: '5%',
  },
}));

export const HomePage = () => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Container className={classes.containerItems}>
        <Paper elevation={3} className={classes.paperLayout}>
          <Typography className={classes.title} component="h3" variant="h2">
            Feedr
          </Typography>
          <TextField
            margin="normal"
            required
            id="email"
            label="Email Address"
            name="email"
            autoComplete="email"
            className={classes.textInput}
          />
          <TextField
            margin="normal"
            required
            name="password"
            label="Password"
            type="password"
            id="password"
            autoComplete="current-password"
            className={classes.textInput}
          />
          <Button className={classes.buttonLogin} component={Link} to={'/feed'}>
            Login
          </Button>
        </Paper>
      </Container>
    </div>
  );
};
