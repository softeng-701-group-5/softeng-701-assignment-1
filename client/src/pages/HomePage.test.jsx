import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { HomePage } from './HomePage';

describe('<HomePage />', () => {
  let subject;

  beforeEach(() => {
    subject = mount(<HomePage />);
  });

  afterEach(() => {
    subject.unmount();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HomePage />, div);
  });
});
