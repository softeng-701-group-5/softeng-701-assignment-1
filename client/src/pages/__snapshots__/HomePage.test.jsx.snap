// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-loginButton-1"
      href="/feed"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          LOGIN
        </p>
      </span>
    </a>
  </div>
</div>
`;
